apisix/plugin.lua

```diff
@@ -228,13 +228,34 @@ function _M.api_routes()
 end
 
 
-function _M.filter(user_route, plugins)
+local function set_response_header_by_debug_flag(plugins, dry_run)
+    if dry_run then
```

Comment:
I think we can use `ngx.headers_sent` instead of passing flag manually.

Reply:
And to avoid the conflict between global rule / route rule, it would be better to use `add_header`.
Maybe we still need to pass the `global` flag to indicate if the current filtering rule is global or not?

Reply:
> I think we can use `ngx.headers_sent` instead of passing flag manually.

yes, this is a much better way. I gonna close this PR.